cmd/oras/root/login.go

```diff
@@ -133,24 +133,21 @@ func runLogin(opts loginOptions) (err error) {
 
 func readLine(prompt string, silent bool) (string, error) {
 	fmt.Print(prompt)
-	if silent {
-		fd := os.Stdin.Fd()
-		state, err := term.SaveState(fd)
-		if err != nil {
-			return "", err
-		}
-		term.DisableEcho(fd, state)
-		defer term.RestoreTerminal(fd, state)
+	fd := int(os.Stdin.Fd())
+	var bytes []byte
+	var err error
+	if silent && term.IsTerminal(fd) {
+		bytes, err = term.ReadPassword(fd)
+		fmt.Println(string(bytes))
+	} else {
+		bytes, err = io.ReadLine(os.Stdin)
+		fmt.Println(string(bytes))
 	}
```

Comment:
`bytes` should be hidden and not printed.
```suggestion
	fd := int(os.Stdin.Fd())
	var bytes []byte
	var err error
	if silent && term.IsTerminal(fd) {
		bytes, err = term.ReadPassword(fd)
	} else {
		bytes, err = io.ReadLine(os.Stdin)
	}
```